### PR TITLE
Getting demo app for vue 3 running

### PR DESCRIPTION
**Pull request by bikingbadger**
Trying to get the Demo running after running into an issue trying to understand the [Real World Example](https://lmiller1990.github.io/vue-testing-handbook/v3/simulating-user-input.html#a-real-world-example) usage of `this.$http`, it seems like the changes to Vue 3 have not been taken into account. This is a PR to try and get that running and working.

### CODE REVIEW

**Comment by webistomin** [comment]
@bikingbadger, seems like you create a new branch from your other pull request. Ideally, there should be a clean branch for every pull request. You will most likely have to resolve conflicts when that pull request is accepted.

File: demo-app-vue-3/package.json [hunk 1]
```diff
@@ -5,10 +5,11 @@
   "scripts": {
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build",
-    "test:unit": "vue-cli-service test:unit"
+    "test:unit": "vue-cli-service test:unit --watch"
```
Comment (lmiller1990) [inline demo-app-vue-3/package.json:8]: Do we need `--watch` here? I think not having the `watch` option makes more sense as the default.

Comment (bikingbadger) [reply demo-app-vue-3/package.json:8]: I will remove it. And do another push.

**Comment by bikingbadger** [comment]
> @bikingbadger, seems like you create a new branch from your other pull request. Ideally, there should be a clean branch for every pull request. You will most likely have to resolve conflicts when that pull request is accepted.

You're correct. I made the last on the fly and wasn't thinking that I would actually make any extra changes. I had done that from the Github interface. I've now fallen down the rabbit hole ;)

**Comment by bikingbadger** [comment]
> Can't believe I missed this - I am sure this was passing locally, weird. We should set up GH actions to run these pre-commit at some point.

I also don't think it's actually running the demo in vue 3 because I stumbled across this trying to figure out the `this.$http` call and couldn't see how it was working. If you don't mind I would like to port a demo over using Vite. Good practice for me and could be a useful source. Especially getting testing setup in Vite which I am currently struggling with

**Comment by lmiller1990** [comment]
Is all this content in https://github.com/lmiller1990/vue-testing-handbook/pull/184? Should we close this in favor of that one?

**Pull request closed without merging**